pkg/queue/workers/task_worker_test.go

```diff
@@ -98,41 +94,6 @@ func TestTaskWorkerWork(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
 	defer cancel()
 
-	t.Run("worker stops and returns error when queue returns a heartbeat error", func(t *testing.T) {
-		qCh := make(chan *queue.Task, 1)
-		q := &mockQueue{queue: qCh, heartbeatErr: errors.New("can not hearbeat")}
-
-		testTask := &queue.Task{
-			TaskBase: queue.TaskBase{
-				Queue: "testQueue",
-			},
-			ID: "testTask",
-		}
-		qCh <- testTask
-
-		handler := queue.TaskHandlerFunc(func(ctx context.Context, task queue.Task, heartbeats chan<- queue.Progress) error {
-			defer close(heartbeats)
-
-			require.NotNil(t, task)
-			require.Equal(t, testTask.ID, task.ID)
-			require.Equal(t, testTask.Queue, task.Queue)
-
-			select {
-			case <-ctx.Done():
-				return nil
-			case <-time.NewTimer(time.Second).C:
-				// mimic a successfull processing
-				heartbeats <- queue.Progress{}
-			}
-
-			return nil
-		})
-
-		w := NewTaskWorker(q, handler)
-		err := w.Work(ctx)
-		require.EqualError(t, err, "can not hearbeat")
-	})
-
```

Comment:
This is not true anymore, the worker continues working after an error during in handling of a single task. It waits for a set timeout until the next try.